OfficeDev/office-ui-fabric-react
# Convert ContextualMenu to CSS in JS

#### Pull request checklist

Marked as **minor** bump.

- [x] Addresses an existing issue: Fixes #4169

#### Description of changes

This PR (hopefully) finishes the refactor of `ContextualMenu` and `ContextualMenuItem` to use the CSS in JS approach of `mergeStyles` building off of #5356 and #5461. 

**List of Changes:**

- Move interfaces to their appropriate `*.types.ts` file for consistency i.e. `IContextualMenuClassNames` now resides in `ContextualMenuItem.types.ts`.
- Exports `IContextualMenuClassNames ` and `IMenuItemClassNames` as part of fixing #4169 and marks them as `@deprecated`.
- Mark `get*ClassNames` methods as `@deprecated`.
- Condense the `getStyles` call for each component into a single method per component.
- `ContextualMenuItemBase` item render functions refactored from `static` to `private` instance methods to access processed class names.
- Use `getGlobalClassNames` for `getStyle` methods.
- Deprecate in-line `IContextualMenuItem `'s `style` prop usage and suggest `mergeStyles` API.
- Refactor redundant `IMenuItemStyles` into `IContextualMenuItemStyles`.
- Update snapshots with latest enzyme output.

#### Focus areas to test

- Go thru the `ContextualMenu` examples and make sure each variant renders as it does in production.
- Check that the public API has not regressed particularly w.r.t styling.
- Try customizing `ContextualMenu` and `ContextualMenuItem` components using `styles` props. I've added tests confirming the now deprecated approaches continue to apply styles when used instead of `mergeStyles`

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/5885)
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/5942)
